pkg/storageincentives/agent.go

```diff
@@ -387,7 +406,17 @@ func (a *Agent) play(ctx context.Context, round uint64) (uint8, []byte, error) {
 		return 0, nil, err
 	}
 
-	sample, err := a.sampler.ReserveSample(ctx, salt, storageRadius, uint64(timeLimiter))
+	nextRoundNumber := (block / a.blocksPerRound) + 1
+	nextRoundBeginningBlock := nextRoundNumber * a.blocksPerRound
+
+	a.latestPriceMtx.Lock()
+
+	difference := nextRoundBeginningBlock - a.latestPriceBlock
+	minimumBalance := new(big.Int).Add(a.latestTotalPayout, new(big.Int).Mul(a.latestPrice, big.NewInt(int64(difference))))
+
+	a.latestPriceMtx.Unlock()
+
+	sample, err := a.sampler.ReserveSample(ctx, salt, storageRadius, uint64(timeLimiter), minimumBalance)
```

Comment:
it'd be cleaner to just pass the list of batchIDs to ignore, instead of minimum balance